src/models/activity.ts

```diff
@@ -122,14 +122,27 @@ export class ActivityEntity {
           const subscaleTableData = subscaleItem.subscaleTableData
 
           const subscaleTableDataItem = subscaleTableData.find(({ sex, age, rawScore }) => {
-            let reportedAge: number | null = null
-            if (typeof ageAnswer === 'string') {
-              reportedAge = Number(ageAnswer)
-            } else if ('value' in ageAnswer && typeof ageAnswer.value === 'number') {
-              reportedAge = ageAnswer.value
+            let reportedAge: string | null = null
+            if (
+              typeof ageAnswer === 'string' ||
+              ('value' in ageAnswer && ['number', 'string'].includes(typeof ageAnswer.value))
+            ) {
+              reportedAge = String(ageAnswer)
+            }
```

Comment:
`reportedAge = String(ageAnswer)` won't work if `ageAnswer` is the dropdown value object and the age is in  `ageAnswer.value`.  The following should fix it:
```suggestion
            if (typeof ageAnswer === 'string') {
              reportedAge = ageAnswer
            } else if (ageAnswer && 'value' in ageAnswer && ['number', 'string'].includes(typeof ageAnswer.value)) {
              reportedAge = String(ageAnswer.value)
            }
```

Reply:
Thanks, this is actually what I meant to do 🤦‍♂️